server/src/main/java/io/crate/user/UserLookup.java

```diff
@@ -33,6 +33,12 @@ public interface UserLookup {
     @Nullable
     User findUser(String userName);
 
+    /**
+     * finds a user by OID
+     */
+    @Nullable
+    User findUser(Integer userOid);
+
```

Comment:
We could add a default impl for `findUser(String userName)`. This would avoid to create UserStub classes.
```
default User findUser(String userName) {
    return findUser(OidHash.userOid(userName));
}
```

Reply:
Thx, reverted back almost all changes from the second commit, less changes in total now.

In some tests it's important that signature with `String` is abstract : we use 
`User::of` in `testSettingUserIfHttpHeaderNotPresent` and `this::userLookup` in `UserAuthenticationMethodTest` where `userLookup` is local function with `String` argument -> and as I tried to revert changes from the second commit back to master version, an make less changes in irrelevant to this PR files, I went with 
```
@Nullable
default User findUser(Integer userOid) {
    return findUser(userOid.toString());
}
```
done in https://github.com/crate/crate/pull/12553/commits/44e48a77c2b6c1ee0193db05a524195a3eba03d7

Reply:
> will never work

oh, yes, thanks. Pushed a fixup.